domenetjenester/iay/src/main/java/no/nav/foreldrepenger/abakus/iay/InntektsmeldingerTjeneste.java

```diff
@@ -26,11 +36,38 @@ public InntektsmeldingerTjeneste(InntektArbeidYtelseRepository repository) {
         this.repository = repository;
     }
 
+    public InntektsmeldingerDto hentAlleInntektsmeldingerForSak(AktørId aktørId, Saksnummer saksnummer, YtelseType ytelseType) {
+        var grunnlag = repository.hentAlleInntektArbeidYtelseGrunnlagFor(aktørId, saksnummer, ytelseType, false);
+        List<InntektsmeldingDto> inntektsmeldinger = mapUnikeInntektsmeldinger(grunnlag);
+        InntektsmeldingerDto inntektsmeldingerDto = new InntektsmeldingerDto();
+        inntektsmeldingerDto.medInntektsmeldinger(inntektsmeldinger);
+        return inntektsmeldingerDto;
+    }
+
     public ArbeidsforholdInformasjon hentArbeidsforholdInformasjonForKobling(KoblingReferanse koblingReferanse) {
         return repository.hentArbeidsforholdInformasjonForBehandling(koblingReferanse).orElseGet(ArbeidsforholdInformasjonEntitet::new);
     }
 
     public GrunnlagReferanse lagre(KoblingReferanse koblingReferanse, ArbeidsforholdInformasjonBuilder informasjonBuilder, List<Inntektsmelding> alleInntektsmeldinger) {
         return repository.lagre(koblingReferanse, informasjonBuilder, alleInntektsmeldinger);
     }
+
+    private ArbeidsforholdInformasjon getArbeidsforholdInformasjon(InntektArbeidYtelseGrunnlag grunnlag) {
+        return grunnlag.getArbeidsforholdInformasjon()
+            .orElseThrow(() -> new IllegalStateException("Mangler ArbeidsforholdInformasjon i grunnlag (påkrevd her): " + grunnlag.getGrunnlagReferanse()));
+    }
+
+    private List<InntektsmeldingDto> mapUnikeInntektsmeldinger(List<InntektArbeidYtelseGrunnlag> grunnlag) {
+        return grunnlag.stream().flatMap(iayg ->
+            iayg.getInntektsmeldinger()
+                .stream()
+                .map(InntektsmeldingAggregat::getAlleInntektsmeldinger)
+                .flatMap(Collection::stream)
+                .map(im -> {
+                    var mapper = new MapInntektsmeldinger.MapTilDto(getArbeidsforholdInformasjon(iayg));
+                    return mapper.mapInntektsmelding(im);
+                })
+        ).distinct().collect(Collectors.toList());
```

Comment:
Er det rett med distinct her?

Reply:
Tanken er at vi ikkje har lyst å hente ut samme inntektsmelding fleire ganger for ein sak.

Reply:
Siden dtoen ikke har noe equals så vil det ikke gi noen effekt med mindre det er nøyaktige det samme objektet som ligger i lista. Det bør sannsynligvis filtreres på et tidligere tidspunkt.

F.eks ved å skrive en spørring som henter opp dette direkte istedenfor å gå på grunnlagene. Det ville nok vært en god optimalisering